### PR TITLE
FLEX-4998 retrieve light measurements events

### PR DESCRIPTION
**Pull request by robindenadel**

### CODE REVIEW

**Comment by CLAAdmin** [comment]
[CLA assistant check](http://cla.opensmartgridplatform.org:80/OSGP/open-smart-grid-platform?pullRequest=103) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/>

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    public XMLGregorianCalendar convertFrom(final ZonedDateTime source,
+            final Type<XMLGregorianCalendar> destinationType, final MappingContext context) {
+        if (source == null) {
+            return null;
+        }
+
+        try {
+            final GregorianCalendar gregorianCalendar = GregorianCalendar.from(source);
+            return DatatypeFactory.newInstance().newXMLGregorianCalendar(gregorianCalendar);
+        } catch (
```
Comment (smvdheijden) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java]: Code formatting? I believe we have our code formatting configured to have less new lines in a catch statement.

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public ZonedDateTime convertTo(final XMLGregorianCalendar source, final Type<ZonedDateTime> destinationType,
+            final MappingContext context) {
+        if (source == null) {
+            return null;
+        }
+
+        final ZoneId zoneId = ZoneId.of("Europe/Paris");
```
Comment (smvdheijden) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java]: I'm wondering, shouldn't we use UTC in OSGP?

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        try {
+            final GregorianCalendar gregorianCalendar = GregorianCalendar.from(source);
+            return DatatypeFactory.newInstance().newXMLGregorianCalendar(gregorianCalendar);
+        } catch (
+            final DatatypeConfigurationException e
+        ) {
+            LOGGER.debug("newXMLGregorianCalendar failed", e);
+            // Sonar will complain about not rethrowing the exception,
```
Comment (smvdheijden) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java]: I believe sonar will not complain, as long as the exception is either logged or rethrown?
In this case the exception is logged, so I think this comment could be removed?

File: osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/XMLGregorianCalendarToZonedDateTimeConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        try {
+            final ZonedDateTime dateTime = ZonedDateTime.now();
+            final XMLGregorianCalendar xmlGregorianCalendar = DatatypeFactory.newInstance()
+                    .newXMLGregorianCalendar(GregorianCalendar.from(dateTime));
+
+            // Try to map to XML version.
+            final XMLGregorianCalendar mappedXMLGregorianCalendar = this.mapper.map(dateTime,
+                    XMLGregorianCalendar.class);
+            Assert.assertEquals(xmlGregorianCalendar, mappedXMLGregorianCalendar);
+        } catch (
```
Comment (smvdheijden) [inline osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/XMLGregorianCalendarToZonedDateTimeConverterTest.java]: Code formatting?

Comment (smvdheijden) [reply osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/XMLGregorianCalendarToZonedDateTimeConverterTest.java]: Wouldn't it be sufficient to remove the try catch and add a "throws Exception" to the method?

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        if (source == null) {
+            return null;
+        }
+
+        final ZoneId zoneId = ZoneId.of("Europe/Paris");
+        return source.toGregorianCalendar().toZonedDateTime().toLocalDateTime().atZone(zoneId);
+    }
+
+    @Override
+    public boolean canConvert(final Type<?> sourceType, final Type<?> destinationType) {
```
Comment (smvdheijden) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java]: Do we need to override the canConvert method?

Comment (robindenadel) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java]: tests will fail if I remove this method

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java (hunk 1 above)
Comment (robindenadel) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java]: done

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java (hunk 3 above)
Comment (robindenadel) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/mappers/XMLGregorianCalendarToZonedDateTimeConverter.java]: done

File: osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/XMLGregorianCalendarToZonedDateTimeConverterTest.java (hunk 4 above)
Comment (robindenadel) [reply osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/XMLGregorianCalendarToZonedDateTimeConverterTest.java]: done

**Pull request merged**